lib/filterx/filterx-scope.c

```diff
@@ -250,23 +250,40 @@ filterx_scope_register_variable(FilterXScope *self,
 gboolean
 filterx_scope_foreach_variable_readonly(FilterXScope *self, FilterXScopeForeachFunc func, gpointer user_data)
 {
-  FilterXVariable *variables = _get_variable_array(self);
+  gboolean result = TRUE;
+  GHashTable *already_found = g_hash_table_new(g_direct_hash, g_direct_equal);
```

Comment:
Using a hash table here should be fine, vars() shouldn't be used for performance intensive code.

I was thinking how this could be avoided, but no good ideas.

Reply:
The arrays are sorted, they could be merged using a heap, but I think this looks simpler this way.